### PR TITLE
Asset check tests all resources for missing asset helper (#32)

### PR DESCRIPTION
**Pull request by realthargor**
refs #32

- previously only default.hbs was checked

### CODE REVIEW

**Comment by kirrg001** [comment]
I agree, you can use the asset help in other template files. It makes sense to not just scan `default.hbs`.

e.g. casper is using the asset helper [in the error template](https://github.com/TryGhost/Casper/blob/2.3.3/error.hbs#L15).

File: lib/checks/030-assets.js [hunk 1]
```diff
@@ -4,25 +4,31 @@ var _ = require('lodash'),
...
-        while ((assetMatch = assetRegex.exec(defaultHbs.content)) !== null) {
-            failures.push({ref: assetMatch[2]});
+    var assetRegex = /(src|href)=['"](.*?\/assets\/.*?)['"]/gmi,
+        assetMatch, stats, ruleCode, failures;
+
+    // Use Asset helper
+    ruleCode = 'GS030-ASSET-REQ';
+    failures = [];
+
+    theme.files.forEach(function (theme) {
```
[the remark this answers is not in the document]
Comment (realthargor) [reply lib/checks/030-assets.js]: When using `theme.templates.all` test failed and I could not find a initialization for this.
Should be in readTheme?
I implemented a filter like in other places to just scan .hsb instead.

[the remark this answers is not in the document]
Comment (kirrg001) [reply lib/checks/030-assets.js]: Ok 👍

File: lib/checks/030-assets.js [hunk 2]
```diff
@@ -4,25 +4,30 @@ var _ = require('lodash'),
...
-        failures = [],
-        defaultHbs = _.filter(theme.files, {file: 'default.hbs'}),
-        assetRegex = /(src|href)=['"](.*?\/assets\/.*?)['"]/gmi,
-        assetMatch, stats;
-
-    if (!_.isEmpty(defaultHbs)) {
-        defaultHbs = defaultHbs[0];
-        while ((assetMatch = assetRegex.exec(defaultHbs.content)) !== null) {
-            failures.push({ref: assetMatch[2]});
+    var assetRegex = /(src|href)=['"](.*?\/assets\/.*?)['"]/gmi,
```
Comment (kirrg001) [inline lib/checks/030-assets.js]: Could you please use `let` or `const`? Thanks 👍

**Comment by kirrg001** [comment]
Will review again asap. Thanks so much 👍

**Comment by gargol** [comment]
Was superseded by https://github.com/TryGhost/gscan/pull/226

**Pull request closed without merging**